iree/compiler/Dialect/Flow/Transforms/FoldCompatibleDispatchRegions.cpp

```diff
@@ -203,14 +203,35 @@ bool isDispatchRegionMergable(DispatchRegionOp &regionOp) {
       // TODO(b/144530470): replace with tablegen attributes/interfaces.
       if (isa<mhlo::ConcatenateOp, mhlo::ConvOp, mhlo::DotGeneralOp,
               mhlo::DotOp, mhlo::PadOp, mhlo::ReduceOp, mhlo::ReduceWindowOp,
-              mhlo::SliceOp, mhlo::TorchIndexSelectOp>(op)) {
+              mhlo::SliceOp>(op)) {
         return false;
       }
     }
   }
   return regionOp.body().getBlocks().size() == 1;
 }
 
+// Returns true if rhs has ops that can only be root op and will lose the
+// characteristic if merge two dispatch regions.
+bool rhsHasRootOnlyOp(DispatchRegionOp &lhs, DispatchRegionOp &rhs) {
+  auto &rhsBlock = rhs.body().front();
+  auto lhsArgs = llvm::to_vector<8>(lhs.args());
+  auto rhsArgs = llvm::to_vector<8>(rhs.args());
+  for (int rhsOpIdx = 0; rhsOpIdx < rhsArgs.size(); ++rhsOpIdx) {
+    for (int lhsResultIdx = 0; lhsResultIdx < lhs.getNumResults();
+         ++lhsResultIdx) {
+      if (rhsArgs[rhsOpIdx] == lhs.getResult(lhsResultIdx)) {
+        for (auto *user : rhsBlock.getArgument(rhsOpIdx).getUsers()) {
+          if (isa<mhlo::TorchIndexSelectOp>(user)) {
```

Comment:
it'd be really nice to remove as many of these hardcoded ops as possible - is this still actually required to be inspected manually like this? can the trait you're looking for here be generalized (like isIndexableOp in dispatch analysis, etc) so that we have a chance of not ending up with an explosion of things in this file?

Reply:
if this kind of super specific check is needed, move it to DispatchConfig.h as an `isFoo(op)` method

Reply:
I plan to move it to DispatchConfig.h

For long-term, we would like to remove most of these limits, but now I'm trying to push the boundary so we will get more ideas where we are and how to push things forward.

Reply:
SGTM! what you're doing now is super useful as it lets us know which ops we actually care about and hopefully come up with some unique traits for them that we can upstream and apply to other frontend dialects like TOSA - way easier to do that when we have the minimal set and not a bunch of random ones :)